activesupport/lib/active_support/core_ext/hash/indifferent_access.rb

```diff
@@ -20,4 +20,14 @@ def with_indifferent_access
   #   { a: b }.with_indifferent_access['a'] # calls b.nested_under_indifferent_access
   #   # => {"b"=>1}
   alias nested_under_indifferent_access with_indifferent_access
+
+  alias_method :original_eql?, :==
```

Comment:
We can use `Module.prepend` here instead of `alias_method`. Could you change that?